hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -127,6 +126,28 @@ public ConfigurationSource getConf() {
    */
   @Override
   public List<DatanodeDetails> chooseDatanodes(
+          List<DatanodeDetails> excludedNodes, List<DatanodeDetails> favoredNodes,
+          int nodesRequired, long metadataSizeRequired, long dataSizeRequired)
+          throws SCMException {
+    return chooseDatanodesInternal(
+            excludedNodes.stream()
```

Comment:
Probably need a comment here explaining why we are doing the transformation on the excludedNodes.